third_party/move/tools/move-mutator/src/compiler.rs

```diff
@@ -43,20 +51,158 @@ pub fn generate_ast(
     config: &BuildConfig,
     package_path: &Path,
 ) -> Result<(FilesSourceText, move_compiler::typing::ast::Program), anyhow::Error> {
-    let mut source_files = mutator_config
+    trace!("Generating AST for package: {package_path:?} and config: {config:?}");
+
+    let source_files = mutator_config
         .project
         .move_sources
         .iter()
         .map(|p| p.to_str().expect("source path contains invalid characters"))
         .collect::<Vec<_>>();
 
-    // If -m option is specified we should use only `move_sources`. However, if `move_sources` is empty
-    // we should add a package path. This path should be always set by the calling function.
-    // In case of any error we should use current directory as package root.
-    if source_files.is_empty() {
-        source_files.push(package_path.to_str().unwrap_or("."));
+    // If -m option is specified we should use only `move_sources`. Using Move source means we won't
+    // check for deps or resolve names as there might be no standard package layout. That means we can mutate
+    // only quite simple files.
+    let compiler = if source_files.is_empty() {
+        prepare_compiler_for_package(config, package_path)?
+    } else {
+        prepare_compiler_for_files(config, source_files)
+    };
+
+    let (files, res) = compiler.run::<PASS_TYPING>()?;
+
+    let (_, stepped) = unwrap_or_report_diagnostics(&files, res);
+    let (_, ast) = stepped.into_ast();
+
+    trace!("Sources parsed successfully, AST generated.");
```

Comment:
nit: I'm not sure if we use dots or not in traces - check the trace above in this function
```suggestion
    trace!("Sources parsed successfully, AST generated");
```

Reply:
Done.